code/modules/mob/living/silicon/ai/ai.dm

```diff
@@ -54,7 +54,7 @@
 	var/list/datum/ai_module/current_modules = list()
 	var/can_dominate_mechs = FALSE
 	var/shunted = FALSE //1 if the AI is currently shunted. Used to differentiate between shunted and ghosted/braindead
-
+	var/obj/machinery/ai_voicechanger/ai_voicechanger = null // reference to machine that holds the voicechanger
```

Comment:
you need to handle this being deleted or use a weakref

Reply:
If this gets deleted nothing happens, i might be too dumb to notice where it would break something. Unless you VV remove the machine without removing UI action, in that case it runtimes.

Reply:
Kyler's right, you need to clear these refs in the destroy of both